frankenstein/cmd/frankenstein/main.go

```diff
@@ -86,7 +86,9 @@ func main() {
 	case distributor:
 		setupDistributor(consul, consulPrefix, chunkStore, remoteTimeout)
 	case ingestor:
-		setupIngestor(consul, consulPrefix, chunkStore, listenPort)
+		ingestor := setupIngestor(consul, consulPrefix, chunkStore, listenPort)
+		defer deleteIngestorConfigFromConsul(consul, consulPrefix)
+		defer ingestor.Stop()
```

Comment:
You mean they'll be run? Odd cause when I try it locally with this and send SIGTERM, it doesn't print "EXITING":

```go
package main

import "fmt"

func foo() {
	fmt.Println("EXITING")
}

func main() {
	defer foo()
	for {
	}
}
```

What's different here?